multicluster/service-mirror/probe_worker.go

```diff
@@ -27,6 +29,7 @@ type ProbeWorker struct {
 func NewProbeWorker(localGatewayName string, spec *multicluster.ProbeSpec, metrics *ProbeMetrics, probekey string) *ProbeWorker {
 	return &ProbeWorker{
 		localGatewayName: localGatewayName,
+		Liveness:         make(chan bool),
```

Comment:
consider making this a buffered channel.  that way the ProbeWorker won't get blocked waiting for the cluster watcher to consume the liveness update.

Reply:
Good call. I gave it a buffer of `10`. Since we're dealing with liveness which is determined over a network, a buffer of `1` felt a little too small to deal with quick changes in status that may happen for network or gateway reasons. That way we're not blocked after cluster watcher watches falls behind by 1 update.